shared/lib_irradproc.cpp

```diff
@@ -2086,6 +2145,37 @@ void irrad::set_sun_component(size_t index, double value) {
     }
 }
 
+void irrad::set_from_weather_record(weather_record wf, weather_header hdr, std::vector<double>& monthlyTiltDegrees, bool useWeatherFileAlbedo,
+             std::vector<double>& userSpecifiedAlbedo, poaDecompReq *poaAllIn, bool useSpatialAlbedos, const util::matrix_t<double>* userSpecifiedSpatialAlbedos, 
+             bool useCustomRotAngles, double customRotAngle) {
+    set_time(wf.year, wf.month, wf.day, wf.hour, wf.minute, delt);
+    set_optional(hdr.elev, wf.pres, wf.tdry);
+    if (radiationMode == irrad::DN_DF) set_beam_diffuse(wf.dn, wf.df);
+    else if (radiationMode == irrad::DN_GH) set_global_beam(wf.gh, wf.dn);
+    else if (radiationMode == irrad::GH_DF) set_global_diffuse(wf.gh, wf.df);
+    else if (radiationMode == irrad::POA_R) set_poa_reference(wf.poa, poaAllIn);
+    else if (radiationMode == irrad::POA_P) set_poa_pyranometer(wf.poa, poaAllIn);
+
+    int month_idx = wf.month - 1;
+    if (useWeatherFileAlbedo && std::isfinite(wf.alb) && wf.alb > 0 && wf.alb < 1) {
+        albedo = wf.alb;
+        albedoSpatial.assign(userSpecifiedSpatialAlbedos->ncols(), albedo);
+    }
+    else if (useSpatialAlbedos) {
+        albedoSpatial = userSpecifiedSpatialAlbedos->row(month_idx).to_vector();
+        albedo = std::accumulate(albedoSpatial.begin(), albedoSpatial.end(), 0.) / albedoSpatial.size();
+    }
+    else {
+        albedo = userSpecifiedAlbedo[month_idx];
+        albedoSpatial.assign(userSpecifiedSpatialAlbedos->ncols(), albedo);
+    }
+    if (trackingMode == TRACKING::SEASONAL_TILT) {
+        tiltDegrees = monthlyTiltDegrees[month_idx];
+        trackingMode = TRACKING::FIXED_TILT;
```

Comment:
Because the irrad constructor is now no longer accessing Subarrays[nn]->trackMode in the loop, this line is changing the trackingMode for good, thus locking you out of the if loop check. So I think everything else works as intended in the hash table.